src/Benchmark.elm

```diff
@@ -212,10 +211,19 @@ benchmark function**. For example, use only `benchmark2` instead of mixing
 `benchmark` and `benchmark2`. The difference between the different benchmark
 functions is small, but not so small that it won't influence your results.
 See the chart in the README for more on the runtime cost of different functions.
+
 -}
-compare : String -> Benchmark -> Benchmark -> Benchmark
-compare =
-    Internal.Compare
+compare : String -> List Benchmark -> Benchmark
+compare name benchmarks =
+    case benchmarks of
+        [] ->
+            Internal.Group name []
+
+        [ benchmark ] ->
+            benchmark
```

Comment:
These two cases are why `compare` explicitly takes two benchmarks at the moment. This increases the possibility that the user will use the API incorrectly and get bad results. It sounds like your use case needs `compare3`, not `compareListOfPotentitallyInfiniteOrZeroSize`. It might be better to implement `compare3` instead.